symbology/create-symbols-from-web-styles/src/main/java/com/esri/samples/create_symbols_from_web_styles/CreateSymbolsFromWebStylesSample.java

```diff
@@ -0,0 +1,316 @@
+          new Alert(Alert.AlertType.ERROR, "Feature layer failed to load. Details: \n"
+            + featureLayer.getLoadError().getCause().getMessage()).show();
+        }
+      });
+
+      // create a unique value renderer
+      UniqueValueRenderer uniqueValueRenderer = new UniqueValueRenderer();
+
+      // add the name of a field from the feature layer data that symbols will be mapped to
+      uniqueValueRenderer.getFieldNames().add("cat2");
+
+      // set the unique value renderer on the feature layer
+      featureLayer.setRenderer(uniqueValueRenderer);
+
+      // create a symbol style from a web style
+      symbolStyle = new SymbolStyle("Esri2DPointSymbolsStyle", null);
+
+      // display an error if the symbol style fails to load
+      symbolStyle.addDoneLoadingListener(() -> {
+        if (symbolStyle.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Error: could not load symbol style. Details: \n"
+            + symbolStyle.getLoadError().getMessage()).show();
+        }
+      });
+
+      // setup the UI for the legend
+      setupLegend();
+
+      // create a list of the required symbol names from the web style
+      ArrayList<String> symbolNames = new ArrayList<>();
+      symbolNames.addAll(Arrays.asList("atm", "beach", "campground", "city-hall", "hospital", "library", "park",
+        "place-of-worship", "police-station", "post-office", "school", "trail"));
+
+      for (String symbolName : symbolNames) {
+
+        // search for each symbol in the symbol style
+        List<String> searchKey = new ArrayList<>();
+        searchKey.add(symbolName);
+        ListenableFuture<Symbol> searchResult = symbolStyle.getSymbolAsync(searchKey);
+        searchResult.addDoneListener(() -> {
+          try {
+            // get the symbol from the search result
+            Symbol symbol = searchResult.get();
+
+            // get a list of all categories to be mapped to the symbol
+            List<String> categories = mapSymbolNameToField(symbolName);
+
+            for (String category : categories) {
+
+              // create a unique value for each category
+              List<Object> categoryValue = new ArrayList<>();
+              categoryValue.add(category);
+              UniqueValueRenderer.UniqueValue uniqueValue = new UniqueValueRenderer.UniqueValue(
+                "", symbolName, symbol, categoryValue);
+
+              // add each unique value to the unique value renderer
+              uniqueValueRenderer.getUniqueValues().add(uniqueValue);
+            }
+
+            // create and add an image view and a label for the symbol to the legend on the UI
+            ImageView imageView = createImageView(symbol);
+            Label gridPaneLabel = new Label(symbolName);
+            gridPane.add(imageView, 0, symbolNames.indexOf(symbolName) + 2);
+            gridPane.add(gridPaneLabel, 1, symbolNames.indexOf(symbolName) + 2);
```

Comment:
Maybe a small comment to say here that the image views and labels are added according to the item's position in the list of `symbol names`? Particularly the +2 is a little puzzling, I'm guessing that's where the symbol/name title are located. Might be easier to extract a variable `Int something = symbolNames.indexOf(symbolName) + 2` 
Alternatively maybe have 2 grid panes, one for the table headers `(Label symbolTitle = new Label("Symbol");` and `Label nameTitle = new Label("Name");`, and then one for the 'data'? We can discuss over a call, might be easier to explain

Reply:
I've separated this into 2 `GridPanes`.  I tried to move some styling into a css file, but of course the sample viewer doesn't read the individual sample's style file (until we are able to put some time into that) so the unique styles won't carry over. So for now, I've left all the required styling where it is.

Reply:
Looks much nicer with the two grid panes 👍 
As for the CSS, I wouldn't worry about that too much. As long as it works in the SV then we are good to go.